### DISCUSSION TITLE
Internal links only?

### DISCUSSION THREAD

**Discussion by lucperkins** [lycheeverse/lychee#838] [Q&A] [answered]
Is there a way to use Lychee to check only internal links on a project? I like to separate internal and external link checks (internal link checks should cause a PR to fail, whereas external link checks shouldn't) and I'd like to find a tool that can do both well. Lychee is fantastic for external links and I'd love to support the former use case as well if possible, but I'm not seeing how to configure that. Any help would be greatly appreciated!

**Comment by mre** [comment]
Thanks for the kind words!

It depends on how you differentiate between internal and external links. If internal links are classified by one or more domains, you can try `--include example.com`. Does that help?

**Reply by lucperkins** [reply]
In this case I'm hoping to check internal relative links like `/about`, in which case a domain-based strategy doesn't work. Now, it's entirely possible that this isn't the right tool for that, but I'm hoping it could be 😀

In other words, I want separate strategies, where the internal link check is super fast and the external link check is cached and such.

**Reply by mre** [reply]
You could play around with the `--offline` flag, which prevents all network requests.

**Comment by mre** [comment, marked as answer]
I'm aware that this is an old discussion, but I just realized that the comments at https://github.com/lycheeverse/lychee/discussions/862 could be relevant.

Specifically, a workaround at the moment is to use

```
lychee --dump
```

and then grepping for the links/patterns you want to have checked.
You can use ripgrep for that.

So

```
# Using grep
lychee --dump | grep '^/' > absolute-links.txt

# Using ripgrep 
lychee --dump | rg '^/' > absolute-links.txt
```

and then

```
lychee --verbose absolute-links.txt
```

Hope that helps someone. I wil close this discussion in favor of https://github.com/lycheeverse/lychee/discussions/862.
